Monika After Story/game/script-affection.rpy

```diff
@@ -2106,6 +2088,163 @@ label monika_affection_nickname_bad_lock:
+    m 1eka "Since you can call me by a nickname now, I was wondering if I could call you by a nickname."
+
+    m 1etc "Is that alright with you?{nw}"
+    $ _history_list.pop()
+    menu:
+        m "Is that alright with you?{fast}"
+
+        "Sure, [m_name].":
+            m 1hua "Great!"
+            m 3eud "I should ask though, what names are you comfortable with?"
+            call mas_player_nickname_loop("Diselect the names you're not comfortable with me calling you", base_nicknames)
+
+        "No.":
+            m 1eka "Alright, [player]."
+            m 3eua "Just let me know if you ever change your mind, okay?"
+
+    #Now unlock the nickname change ev
+    $ mas_unlockEVL("monika_change_player_nicknames", "EVE")
+    return "no_unlock"
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_change_player_nicknames",
+            prompt="Can you call me different nicknames?",
+            category=['you'],
+            pool=True,
+            unlocked=False,
+            rules={"no_unlock": None},
+            aff_range=(mas_aff.AFFECTIONATE,None)
+        )
+    )
+
+label monika_change_player_nicknames:
+    m 1hub "Sure [player]!"
+
+    python:
+        #Generate a list of names we're using now so we can set things
+        current_nicknames = [
+            (nickname.capitalize(), nickname, True, True, False)
+            for nickname in persistent._mas_player_nicknames
+        ]
+
+        if not current_nicknames:
+            current_nicknames = [
+                ("Darling", "darling", True, True, False),
+                ("Honey", "honey", True, True, False),
+                ("Love", "love", True, True, False),
+                ("My love", "my love", True, True, False),
+                ("Sweetheart", "sweetheart", True, True, False),
+                ("Sweetie", "sweetie", True, True, False),
+            ]
+
+    call mas_player_nickname_loop("Deselect the names you don't want me to call you anymore.", current_nicknames)
+    return
+
+label mas_player_nickname_loop(check_scrollable_text, nickname_pool):
+    show monika 1eua at t21 zorder MAS_MONIKA_Z with dissolve
+    $ renpy.say(m, renpy.substitute(check_scrollable_text), interact=False)
+    call screen mas_check_scrollable_menu(nickname_pool, mas_ui.SCROLLABLE_MENU_TXT_AREA, mas_ui.SCROLLABLE_MENU_XALIGN)
+
+    $ acceptable_nicknames = _return.keys()
+    $ done = False
+
+    show monika at t11
+    while not done:
+        m 1eua "Is there anything else you'd like me to call you?{nw}"
+        $ _history_list.pop()
+        menu:
+            m "Is there anything else you'd like me to call you?{fast}"
+
+            "Yes.":
+                m 3eua "Sure, just type 'nevermind' if you change your mind."
+
+                label .name_enter_skip_loop:
+                    pass
+
+                #Now parse this
+                python:
+                    lowername = renpy.input(
```

Comment:
don't we have a new `mas_input` screen that has its own nevermind option? Can we use that here instead?